src/Pos/Ssc/GodTossing/Core/Core.hs

```diff
@@ -290,3 +295,32 @@ mkGtProof payload =
       where
         proof constr hm cert =
             constr (hash hm) (hash cert)
+
+-- | Transforms GtPayload to fit under size limit.
+stripGtPayload :: Byte -> GtPayload -> Maybe GtPayload
+stripGtPayload lim payload | biSize payload <= lim = Just payload
+stripGtPayload lim payload = case payload of
+    (CertificatesPayload vssmap) -> CertificatesPayload <$> stripHashMap lim vssmap
+    (CommitmentsPayload (getCommitmentsMap -> comms0) certs0) -> do
+        let certs = stripHashMap limCerts certs0
+        let comms = stripHashMap (lim - biSize certs) comms0
+        CommitmentsPayload <$> (mkCommitmentsMapUnsafe <$> comms) <*> certs
+    (OpeningsPayload openings0 certs0) -> do
+        let certs = stripHashMap limCerts certs0
+        let openings = stripHashMap (lim - biSize certs) openings0
+        OpeningsPayload <$> openings <*> certs
+    (SharesPayload shares0 certs0) -> do
+        let certs = stripHashMap limCerts certs0
+        let shares = stripHashMap (lim - biSize certs) shares0
```

Comment:
Perhaps it makes more sense to prioritize comms/openings/shares rather than certs, but it's minor and I don't insist.